python/tests/test_topology.py

```diff
@@ -7951,3 +7953,163 @@ def test_star_branch_length(self):
 
         assert ts.node(ts.first().root).time == branch_length
         assert ts.kc_distance(topological_equiv_ts) == 0
+
+
+class TestPolytomySplitting(unittest.TestCase):
+    """
+    Test the ability to randomly split polytomies
+    """
+
+    # A complex ts with polytomies - in the first 2 trees the polytomy involves
+    # the same children, so should be resolved in one go
+    #
+    # 1.00┊    6      ┊      6    ┊       6   ┊           ┊      6    ┊
+    #     ┊ ┏━┳┻┳━┓   ┊   ┏━┳┻┳━┓ ┊    ┏━━╋━┓ ┊           ┊   ┏━┳┻┳━┓ ┊
+    # 0.50┊ 5 ┃ ┃ ┃   ┊   5 ┃ ┃ ┃ ┊    5  ┃ ┃ ┊      5    ┊   ┃ ┃ ┃ ┃ ┊
+    #     ┊ ┃ ┃ ┃ ┃ . ┊   ┃ ┃ ┃ ┃ ┊ . ┏┻┓ ┃ ┃ ┊ . ┏━┳┻┳━┓ ┊ . ┃ ┃ ┃ ┃ ┊
+    # 0.00┊ 0 2 3 4 1 ┊ 0 1 2 3 4 ┊ 0 1 2 3 4 ┊ 0 1 2 3 4 ┊ 0 1 2 3 4 ┊
+    #   0.00        0.20        0.40        0.60        0.80        1.00
+    nodes_polytomy_44344 = """\
+    id      is_sample   population      time
+    0       1           0               0.0
+    1       1           0               0.0
+    2       1           0               0.0
+    3       1           0               0.0
+    4       1           0               0.0
+    5       0           0               0.5
+    6       0           0               1.0
+    """
+    edges_polytomy_44344 = """\
+    id      left     right    parent  child
+    0       0.0      0.2      5       0
+    1       0.0      0.8      5       1
+    2       0.0      0.4      6       2
+    3       0.4      0.8      5       2
+    4       0.0      0.6      6       3,4
+    5       0.0      0.6      6       5
+    6       0.6      0.8      5       3,4
+    7       0.8      1.0      6       1,2,3,4
+    """
+
+    def ts_polytomy_4(self):
+        return tskit.Tree.generate_star(4).tree_sequence
+
+    def ts_polytomy_44344(self):
+        return tskit.load_text(
+            nodes=io.StringIO(self.nodes_polytomy_44344),
+            edges=io.StringIO(self.edges_polytomy_44344),
+            strict=False,
+        )
+
+    # Statistical test coded here for reference
+    # @unittest.skip("Testing statistical properties is nontrivial in a unit test")
+    def test_equiprobable(self):
```

Comment:
I don't think we can do this as a unit test as it'll be too slow and cause problems. We'll need some sort of validation suite or something. I'm not sure what to do with the code for now though.

@benjeffery, any thoughts?

Reply:
For the moment I can uncomment the skip and say "To be put in validation suite". How is this sort of simulation testing done in msprime?

Reply:
I didn't know about `@pytest.mark.slow` - seems like a good compromise.